design/bgp-bfd.md

```diff
@@ -0,0 +1,114 @@
+# BFD enablement in BGP
+
+## Summary
+
+The purpose of this enhancement is to use enable [BFD](https://en.wikipedia.org/wiki/Bidirectional_Forwarding_Detection) to be used in conjunction with BGP.
+
+BFD will be available only in the upcoming [FRR based implementation](0001-frr.md).
+
+## Motivation
+
+BFD provides a quicker path failure detection than BGP, and using them together will allow users to provide a better
```

Comment:
Not really a review comment, but just a FYI in case you weren't aware, as it is not very documented IIRC. We have a know to configure the hold-time for BGP peers. The min is 3 seconds (IIRC that was imposed by BGP), and setting that to 3s really improves the failure detection.

You can see it here: https://github.com/metallb/metallb/blob/2671c3f22de729e5faad7b35732bba1314932d0d/internal/config/config.go#L46

Far from the ms failure detection we can get with BFD IIUC, though. But maybe the hold-time can help you to aliviate some issues meanwhile :)

Reply:
A sentence could be added here to say:

`Prior to adding BFD support, you can configure the hold-time parameter to speed up failure detection, but the minimum or fastest value is 3 seconds.`